odoo/api.py

```diff
@@ -987,27 +991,31 @@ def set(self, record, field, value, dirty=False, check_dirty=True):
             dirty must raise an exception
         """
         field_cache = self._set_field_cache(record, field)
+        record_id = record.id
```

Comment:
This part is an optimization of method `set()`, and should be put into a separate commit.